repos/system_upgrade/el7toel8/actors/tcpwrapperscheck/actor.py

```diff
@@ -0,0 +1,64 @@
+from leapp.actors import Actor
+from leapp.models import Report, TcpWrappersFacts, InstalledRedHatSignedRPM
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.common.tcpwrappersutils import config_applies_to_daemon
+from leapp.libraries.common.reporting import report_with_links
+
+DAEMONS = [
+    ("audit", ["auditd"]),
+    ("bacula", ["bacula"]),
+    ("conserver", ["conserver"]),
+    ("cyrus-imapd", ["imap", "pop3", "imaps", "pop3s", "sieve", "lmtp"]),
+    ("dovecot", ["imap", "pop3", "imaps", "pop3s", "sieve", "lmtp"]),
+    ("nfs-utils", ["mountd", "statd"]),
+    ("openssh-server", ["sshd"]),
+    ("proftpd", ["proftpd"]),
+    ("quota", ["rquotad"]),
+    ("rpcbind", ["rpcbind"]),
+    # ("sendmail", ["sendmail"]),  # Handled in CheckSendmail Actor
+    ("slapi-nis", ["slapi-nis"]),
+    ("socat", ["socat"]),
+    ("stunnel", ["stunnel"]),
+    ("tftp-server", ["tftpd"]),
+    # ("vsftpd", ["vsftpd"]),  # Handled in VsftpdConfigCheck Actor
+    ("xinetd", ["xinetd"]),
+]
+
+
+class TcpWrappersCheck(Actor):
+    """
+    Check the list of packages previously compiled with TCP wrappers support
+    and check whether they have some rules configured in
+    /etc/hosts.{allow,deny} that are no longer honored by the daemons and
+    might make the service more accessible than expected.
+    """
+
+    name = 'tcp_wrappers_check'
+    consumes = (TcpWrappersFacts, InstalledRedHatSignedRPM,)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        tcp_wrappers_facts = next(self.consume(TcpWrappersFacts))
+
+        found_packages = set()
+        found_daemons = set()
+        for (package, keywords) in DAEMONS:
+            if has_package(InstalledRedHatSignedRPM, package):
```

Comment:
Not sure if has_package is the right choice here - it will recreate the installed_packages lookup for each package in a cycle.
I'd probably go with manually creating the lookup and using it to check for package presence in a cycle, smth like
```
from leapp.libraries.common.rpms import create_lookup
...

def process(self):
...
installed_packages  = create_lookup(InstalledRedHatSignedRPM, field='items', key='name')
for (package, keywords) in DAEMONS:
    if package in installed_packages:
        ...
```